services/spiget/spiget-base.js

```diff
@@ -0,0 +1,59 @@
+'use strict'
+
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
+
+const schema = Joi.object({
+  downloads: Joi.number().required(),
+
+  file: Joi.object({
+    size: Joi.number().required(),
+    sizeUnit: Joi.string().required(),
+  }).required(),
+
+  testedVersions: Joi.array(),
+
+  rating: Joi.object({
+    count: Joi.number().required(),
+    average: Joi.number().required(),
+  }).required(),
+}).required()
+
+const versionSchema = Joi.object({
+  downloads: Joi.number().required(),
+  name: Joi.string().required(),
+}).required()
+
+const documentation = `
+<p>You can find your resource ID in the url for your resource page.</p>
+<p>Example: <code>https://www.spigotmc.org/resources/essentialsx.9089/</code> - Here the Resource ID is 9089.</p>`
+
+class BaseSpigetService extends BaseJsonService {
+  async fetch({ resourceid }) {
+    const url = `https://api.spiget.org/v2/resources/${resourceid}`
+
+    return this._requestJson({
+      schema,
+      url,
+    })
+  }
+
+  async fetchLatestVersion({ resourceid }) {
```

Comment:
This function is only called from a single service (`spiget-latest-version`) so it appears to be tailored to a very specific use case. Let's try refactoring your `fetch` function a bit to take in additional parameters for `schema` and `url` (you can give them default values of `schema` and `https://api.spiget.org/v2/resources/${resourceid}`). 

You can then remove this function from the base class, move the `versionSchema` and `url` to the `spiget-latest-version` service, and update that service to call `fetch`, and pass the `url` (https://api.spiget.org/v2/resources/${resourceid}/versions/latest) and `versionSchema` as arguments